Assets/HoloToolkit/Sharing/Scripts/Utilities/SharingServiceStatus.cs

```diff
@@ -0,0 +1,61 @@
+﻿using UnityEngine;
+using System.Collections;
+using UnityEngine.UI;
+
+namespace HoloToolkit.Sharing
+{
+    public class SharingServiceStatus : MonoBehaviour
+    {
+        enum Status
+        {
+            Disconnected,
+            Connecting,
+            Connected,
+        }
+
+
+        [Tooltip("Reference to Text UI control where the Status info should be displayed.")]
+        public Text Text;
+
+        [Tooltip("Reference to Image UI control where the Status background color should be applied.")]
+        public Image Background;
+
+        void Start()
+        {
+            SetStatus(Status.Disconnected);
+
+            SharingStage.Instance.SharingManagerConnected += SharingManagerConnected;
+
+            if (SharingStage.Instance.AutoDiscoverServer)
+                SetStatus(Status.Connecting);
+        }
+
+        private void SharingManagerConnected(object sender, System.EventArgs e)
+        {
+            SetStatus(Status.Connected, SharingStage.Instance.ServerAddress);
+        }
+
+        private void SetStatus(Status state, string detail = "")
+        {
+            Background.color = StatusColor(state);
+            Text.text = state.ToString();
+            if (detail != "")
+                Text.text += ": " + detail;
+        }
+
+        private Color StatusColor(Status s)
+        {
+            switch (s)
+            {
+                case Status.Disconnected:
+                    return Color.red;
+                case Status.Connecting:
+                    return new Color(255, 175, 0);
+                case Status.Connected:
+                    return Color.green;
```

Comment:
I like the idea of making these configurable.
If we also make the change to TextMesh, and get rid of the UI Canvas, we need to redesign the way the status color is displayed as now it's just another UI element on the canvas.
I am thinking this could become a 3D primitive (cube or sphere).
The discussion / question is how do we achieve that?
Do we define one material and allow the user to specify a color that just changes the albedo on the material? This would limit the user to just simple color changes.
Or do we allow the user to pick different material for each state. This is more flexible for the user, but we need to add 3 default materials to the project. too much clutter?